tests/snappi_tests/dataplane/test_bgp_device_unisolation.py

```diff
@@ -0,0 +1,223 @@
+@pytest.mark.parametrize("frame_rate", [10])
+@pytest.mark.parametrize("frame_size", [64, 128, 256, 512, 1024, 1518])
+@pytest.mark.parametrize("event_type", ["config_reload", "all_ports_startup", "bgp_container_restart"])
+@pytest.mark.parametrize("unisolation_device_hostname", ['sonic-s6100-dut3'])
+def test_bgp_sessions(
+    duthosts,
+    snappi_api,
+    get_snappi_ports,
+    fanout_graph_facts_multidut,
+    set_primary_chassis,   # noqa: F811, F401
+    create_snappi_config,  # noqa: F811, F401
+    subnet_type,
+    db_reporter,
+    tbinfo,
+    frame_rate,
+    frame_size,
+    event_type,
+    unisolation_device_hostname
+):
+    """
+    Test to check if packets get dropped on injecting fec errors
+    Note: fanout_per_port is the number of fanouts per fron panel port
+          Example: For running the test on 400g fanout mode of a 800g port,
+          fanout_per_port is 2
+    """
+    snappi_extra_params = SnappiTestParams()
+    pytest_assert(
+        subnet_type in ROUTE_RANGES, "Failing test as no route ranges are provided for {}".format(subnet_type)
+    )
+    snappi_extra_params.ROUTE_RANGES = ROUTE_RANGES
+    snappi_ports = get_duthost_bgp_details(duthosts, get_snappi_ports, subnet_type)
+    tx_ports = snappi_ports[::2]
+    rx_ports = snappi_ports[1::2]
+
+    for dut in duthosts:
+        if dut.hostname == unisolation_device_hostname:
+            snappi_extra_params.unisolation_device = dut
+            break
+    pytest_assert(
+        hasattr(snappi_extra_params, 'unisolation_device'),
+        "Failing test as unable to find the unisolation device hostname: {}".format(unisolation_device_hostname)
+    )
+    snappi_extra_params.protocol_config = {
+        "Tx": {
+            "protocol_type": "bgp",
+            "ports": tx_ports,
+            "subnet_type": subnet_type,
+            "is_rdma": False,
+        },
+        "Rx": {
+            "route_ranges": ROUTE_RANGES[subnet_type] * len(rx_ports),
+            "protocol_type": "bgp",
+            "ports": rx_ports,
+            "subnet_type": subnet_type,
+            "is_rdma": False,
+        },
+    }
+    snappi_config, snappi_obj_handles = create_snappi_config(snappi_extra_params)
+    snappi_extra_params.traffic_flow_config = [
+        {
+            "line_rate": frame_rate,
+            "frame_size": frame_size,
+            "is_rdma": False,
+            "flow_name": "bgp_traffic",
+            "tx_names": snappi_obj_handles["Tx"]["ip"],
+            "rx_names": snappi_obj_handles["Rx"]["network_group"],
+        },
+    ]
+    snappi_config = create_traffic_items(snappi_config, snappi_extra_params)
+    get_convergence_for_device_unisolation(
+        duthosts,
+        snappi_api,
+        snappi_config,
+        tx_ports,
+        rx_ports,
+        subnet_type,
+        snappi_obj_handles,
+        event_type,
+        db_reporter,
+        snappi_extra_params
+    )
+
+
+def get_convergence_for_device_unisolation(
```

Comment:
run_convergence_test_for_device_unisolation

also can we reuse the test skeleton for case 1 and 2? only the failure trigger and recover are different.